### ISSUE TITLE
Nexmo does support all GSM-7 characters by default

### ISSUE THREAD

**Issue by abelzx** [Nexmo/nexmo-developer#701]
## Description

The guide for Encoding does not include all GSM-7 characters and the "Try it out" part is not showing the correct information.
https://developer.nexmo.com/messaging/sms/guides/concatenation-and-encoding

All these characters `|^€{}[~]\` are supported in the GSM-7 extension table (an additional ESC control character is required to access those characters) and Unicode is not necessary.

## Related resources

GSM 7-bit default alphabet and extension table (all characters supported when using default encoding type=text)
https://en.wikipedia.org/wiki/GSM_03.38#GSM_7-bit_default_alphabet_and_extension_table_of_3GPP_TS_23.038_.2F_GSM_03.38

## List of supported characters

```
! " # $ % ' ( ) * + , - . / : ; < = > ? @ _ ¡ £ ¥ § ¿ & ¤ §
0 1 2 3 4 5 6 7 8 9
A B C D E F G H I J K L M N O P Q R S T U V W X Y Z
a b c d e f g h i j k l m n o p q r s t u v w x y z
Ä Å Æ Ç É Ñ Ø ø Ü ß Ö à ä å æ è é ì ñ ò ö ù ü Δ Φ Γ Λ Ω Π Ψ Σ Θ Ξ
| ^ € { } [ ] ~ \
```

**Comment by tbedford** [comment]
1) Yes, by default GSM-7 and extended table are supported.
2) Try it Out is not quite right. If you type a '{' character or other character from extended table it will flag that as requiring a type of 'unicode' whereas a type of 'text' is fine. However, not that a character from the extended table does require two characters (ESC + char).
3) There's another less obvious problem. If you type in 160 normal characters everything is fine - they fit on one message in the Try-it-Out as they should. However, if I delete two characters and then type '{' that should also fit into one message of 160 characters. However it doesn't, because the Try-it-Out behaves as if that character was Unicode. Normally in SMS when a Unicode character is detected it will automatically half the number of characters per message part to allow for two bytes per character as per UCS-2. So Try-it-Out is not working correctly.
4) There are other issues here to dig into. For example, what do we do if a UTF-8 message is sent into Nexmo? How do Nexmo libraries handle UTF-8 encoded streams? Are they automatically handled? 
Requires further investigation to get this right.

**Comment by tbedford** [comment]
I've created #725 to deal with issues relating to the Try-it-out problems. There was basically one issue: that characters from GSM extended were incorrectly identified as requiring Unicode message type.

**Comment by tbedford** [comment]
Fixed via PR #723
